miden-tx/src/tests/kernel_tests/test_tx.rs

```diff
@@ -619,76 +619,75 @@ fn test_build_recipient_hash() {
 // FOREIGN PROCEDURE INVOCATION TESTS
 // ================================================================================================
 
-/// Assert that mock account's `get_item` procedure has index 8
 #[test]
-fn test_check_get_item_index() {
-    // keep this code the same as the actual code of the account's `get_item` procedure
-    let get_item_code_source = "
+fn test_fpi_memory() {
+    // Prepare the test data
+    let storage_slots =
+        vec![AccountStorage::mock_item_0().slot, AccountStorage::mock_item_2().slot];
+    let foreign_account_code_source = "
         use.miden::account
-        export.get_item
+
+        export.get_item_foreign
+            # make this foreign procedure unique to make sure that we invoke the procedure of the 
+            # foreign account, not the native one
+            push.1 drop
             exec.account::get_item
-            # => [VALUE, pad(15)]
 
             # truncate the stack
-            movup.8 drop movup.8 drop movup.8 drop
-            # => [VALUE, pad(12)]
+            movup.6 movup.6 movup.6 drop drop drop
+        end
+
+        export.get_map_item_foreign
+            # make this foreign procedure unique to make sure that we invoke the procedure of the 
+            # foreign account, not the native one
+            push.2 drop
+            exec.account::get_map_item
         end
```

Comment:
Question: do we not need to truncate the stack at the end of this procedure?

Reply:
We don't need to truncate the stack because the overall number of elements at the end of the execution even smaller than the number at the beginning: we truncate the stack from 5 elements + pads to 4 elements + pads. So it is just a convenient coincidence. 
```masm
#! Inputs:  [index, KEY]
#! Outputs: [VALUE]
```